pkg/controller/sshd/sshd_controller.go

```diff
@@ -178,9 +169,27 @@ func (r *ReconcileSSHD) Reconcile(request reconcile.Request) (reconcile.Result,
 		}
 	} else {
 		// Request object is being deleted.
-		if controllerutil.ContainsFinalizer(instance, reconcileSSHDFinalizerDNS) && r.route53 != nil {
+		if controllerutil.ContainsFinalizer(instance, reconcileSSHDFinalizerDNS) {
 			r.SetSSHDStatus(instance, "Deleting DNS aliases", cloudingressv1alpha1.SSHDStateFinalizing)
-			if err = r.deleteDNSRecords(); err != nil {
+
+			// fetch the sshd service
+			svc := &corev1.Service{}
+			err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name}, svc)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			err = r.cloudClient.DeleteSSHDNS(context.TODO(), r.client, instance, svc)
+			switch err {
+			case nil:
+				// all good
+			case err.(*cioerrors.LoadBalancerNotFoundError):
+				// couldn't find the new ELB yet
```

Comment:
Can you expand this comment with what it means that the ELB isn't present? I would think that during a delete it means that the ELB hasn't been created yet (and there is such a creation wanting to happen).

Reply:
That's my understanding too. Maybe it was created and then immediately removed, and might have to wait for the ELB to be created before it can be deleted?

I'm fine removing the error check for `LoadBalancerNotFoundError` on deletion if we don't think it can get into that situation, though.